### DISCUSSION TITLE
SLA Reports with Grafana Graphs?

### DISCUSSION THREAD

**Discussion by heinowalther** [sni/thruk#1094] [Q&A]
We have several installations of Thruk based on the OMD dist.
Older ones use pnp for graphing, and a graph is inserted into the SLA reports generated.

On the newer installations of OMD dist, InfluxDB and Grafana are used for graphs, but they are not inserted into the SLA reports... this sadly leaves them almost useless.

I would be great if this could be done, some how?

I haven't tested, but it might be possible to add the pnp call together with influxdb, and maybe it will insert th pnp version of the graf?

So actually maybe a quick and dirty fix could be to build the pnp graph based on the data in InfluxDB, and insert it into the SLA Report?

**Comment by sni** [comment]
When using OMD, all you need to do is use the `host-perf` or `srv-perf` which sets an action url which can be used to export the graphs. If available, they will be used in the reports as well.
However, with recent grafana versions, you might hit #1055, so you have to install npm/chrome and puppeteer.

**Reply by heinowalther** [reply]
I do use host/srv-perf.
I have now tried to install npm chrome and puppeteer.
But it does not generate the graphs... maybe some time need to pass?  (if this is done from the *-perf scripts?)
I'm not on the nightly, but on 2.40-2 / OMD 4.0
Do I have to change anything else in order to make this work?
Also I searched my system for the grafana_export.sh which is mentioned, but it is nowhere to be found... so maybe that is the problem?

**Comment by sni** [comment]
That might be the issue, i added puppeteer support in Nov 2020, so if your version older, it will definitly not work.
